lib/filter-stream.js

```diff
@@ -176,7 +177,7 @@ function FilterStream( blockMap, options ) {
     var buffer = Buffer.concat( this._buffers )
     var offset = 0
 
-    while( offset < buffer.length ) {
+    while( buffer.length - offset >= this.blockSize ) {
       this._pushBlock( buffer.slice( offset, offset + this.blockSize ) )
       offset = offset + this.blockSize
```

Comment:
No particular reason, just a habit from a time when `+=` wasn't optimised well by VMs – prefer the short variant?